internal/charmstore/search.go

```diff
@@ -134,11 +176,16 @@ func (s *Store) UpdateSearchFields(r *router.ResolvedURL, fields map[string]inte
 	return nil
 }
 
-// searchDocFromEntity performs the processing required to convert a mongodoc.Entity
-// to an esDoc for indexing.
+// searchDocFromEntity performs the processing required to convert a
+// mongodoc.Entity and the corresponding mongodoc.BaseEntity to an esDoc
+// for indexing.
 func (s *Store) searchDocFromEntity(e *mongodoc.Entity, be *mongodoc.BaseEntity) (*SearchDoc, error) {
 	doc := SearchDoc{Entity: e}
 	doc.ReadACLs = be.ACLs.Read
+	if !be.Promulgated {
```

Comment:
Perhaps a comment here, justifying the decision to exclude entities with promulgated URLs that don't have the promulgated bit set in their base entities?